metaflow/plugins/cards/card_cli.py

```diff
@@ -432,23 +589,82 @@ def get(
     pathspec,
     hash=None,
     type=None,
+    id=None,
     follow_resumed=False,
 ):
     """
     Get the HTML string of the card based on pathspec.\n
-    The Task pathspec is of the form:\n
-        <runid>/<stepname>/<taskid>\n
+    The pathspec can be of the form:\n
+        - <stepname>\n
+        - <runid>/<stepname>\n
+        - <runid>/<stepname>/<taskid>\n
     """
+    card_id = id
     available_card_paths, card_datastore, pathspec = resolve_card(
         ctx,
         pathspec,
         type=type,
         hash=hash,
+        card_id=card_id,
         follow_resumed=follow_resumed,
     )
     if len(available_card_paths) == 1:
         print(card_datastore.get_card_html(available_card_paths[0]))
     else:
         list_available_cards(
-            ctx, pathspec, available_card_paths, card_datastore, command="get"
+            ctx,
+            pathspec,
+            available_card_paths,
+            card_datastore,
+            command="get",
+        )
+
+
+@card.command()
+@click.argument("pathspec", required=False)
+@card_read_options_and_arguments
+@click.option(
+    "--as-json",
+    default=False,
+    is_flag=True,
+    help="Print all available cards as a JSON object",
+)
+@click.pass_context
+def list(
```

Comment:
It may make sense in these commands to respect the "--quiet" flag which should only print things that are easily machine parseable (at least that's the intent IIRC). This would impact the listing functions above to have a quiet mode in essence and maybe list things simply as opposed to prettily.

Reply:
Currently, we respect `--quiet` in `list` unless it is called with `--as-json`; the same goes for `get`. If I call `card get stepname` then it will print the card in the `stdout`. But calling `card get stepname filename.html` will save that card to the file. The test uses `--quiet` so I need this functionality to print/pass information when calling CLI commands in CliCheck.